ESP8266Interface.cpp

```diff
@@ -472,6 +478,33 @@ void ESP8266Interface::socket_attach(void *handle, void (*callback)(void *), voi
     _cbs[socket->id].data = data;
 }
 
+nsapi_error_t ESP8266Interface::setsockopt(nsapi_socket_t handle, int level,
+        int optname, const void *optval, unsigned optlen)
+{
+    struct esp8266_socket *socket = (struct esp8266_socket *)handle;
+
+    if(socket->connected == true) {// ESP8266 limitation, keepalive needs to be given before connecting
+        return NSAPI_ERROR_UNSUPPORTED;
+    }
+
+    if (level == NSAPI_SOCKET) {
+        switch (optname) {
+            case NSAPI_KEEPALIVE: {
+                if (optlen == sizeof(_keepalives[socket->id])) {
```

Comment:
Test here must be for `sizeof(int)`, as that's how you're about to access it in the next line.

Reply:
ok